Main thread fix

ikbalkaya opened the pull request:
Regression from https://github.com/ably/ably-flutter/pull/291
Closes https://github.com/ably/ably-flutter/issues/429

KacperKluka commented:
@ikbalkaya are you planning to also fix that ANR issue that is being reintroduced to the SDK with this PR? It would not be good to make a new release that deliberately adds a bug that we've fixed before :sweat_smile:

ikbalkaya commented:
> @ikbalkaya are you planning to also fix that ANR issue that is being reintroduced to the SDK with this PR? It would not be good to make a new release that deliberately adds a bug that we've fixed before 😅

Yes, it is intended to be resolved when we decouple push notification code from library code as I added in description https://github.com/ably/ably-flutter/issues/339 But for now this bug takes the priority on ANR bug so we want to fix this. More discussion on  [this internal message](https://ably-real-time.slack.com/archives/C0178QGC093/p1655718602091409)

KacperKluka reviewed (approved):
LGTM :+1:
